infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -42,51 +42,205 @@ param subnetId string
 @description('Specifies the SKU for the Function App hosting plan')
 param sku object
 
-var kind = 'functionapp'
+@description('Specifies the Key Vault name that this Function App will be permitted to get and list secrets from')
+param keyVaultName string
+
+@description('Specifies whether or not the Function App already exists. This is used to determine whether or not to look for existing appsettings')
+param functionAppExists bool
+
+@description('Specifies the number of pre-warmed instances for this Function App - must be compatible with the chosen hosting plan')
+param preWarmedInstanceCount int?
+
+@description('Specifies the number of pre-warmed instances for this Function App - must be compatible with the chosen hosting plan')
+param alwaysOn bool?
```

Comment:
Done ta!  Also I've just switched back to using `set -e` rather than `failOnStdErr` in the YML task definitions, as they're failing all over the shop for non-failure conditions!